modules/openapi-generator/src/main/resources/typescript-rxjs/runtime.mustache

```diff
@@ -0,0 +1,199 @@
+    private clone<T extends BaseAPI>(this: T): T {
+        const constructor = this.constructor as any;
+        const next = new constructor(this.configuration);
+        next.middleware = this.middleware.slice();
+        return next;
+    }
+}
+
+export class RequiredError extends Error {
+    name: 'RequiredError' = 'RequiredError';
+    constructor(public field: string, msg?: string) {
+        super(msg);
+    }
+}
+
+export const COLLECTION_FORMATS = {
+    csv: ',',
+    ssv: ' ',
+    tsv: '\t',
+    pipes: '|',
+};
+
+export interface ConfigurationParameters {
+    basePath?: string; // override base path
+    middleware?: Middleware[]; // middleware to apply before/after rxjs requests
+    username?: string; // parameter for basic security
+    password?: string; // parameter for basic security
+    apiKey?: string | ((name: string) => string); // parameter for apiKey security
+    accessToken?: string | ((name: string, scopes?: string[]) => string); // parameter for oauth2 security
+}
+
+export class Configuration {
+    constructor(private configuration: ConfigurationParameters = {}) {}
+
+    get basePath(): string {
+        return this.configuration.basePath || BASE_PATH;
+    }
+
+    get middleware(): Middleware[] {
+        return this.configuration.middleware || [];
+    }
+
+    get username(): string | undefined {
+        return this.configuration.username;
+    }
+
+    get password(): string | undefined {
+        return this.configuration.password;
+    }
+
+    get apiKey(): ((name: string) => string) | undefined {
+        const apiKey = this.configuration.apiKey;
+        if (apiKey) {
+            return typeof apiKey === 'function' ? apiKey : () => apiKey;
+        }
+        return undefined;
+    }
+
+    get accessToken(): ((name: string, scopes?: string[]) => string) | undefined {
+        const accessToken = this.configuration.accessToken;
+        if (accessToken) {
+            return typeof accessToken === 'function' ? accessToken : () => accessToken;
+        }
+        return undefined;
+    }
+}
+
+export type Json = any;
+export type HttpMethod = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE' | 'OPTIONS';
+export type HttpHeaders = { [key: string]: string };
+export type HttpQuery = { [key: string]: string | number | null | boolean | Array<string | number | null | boolean> };
+export type HttpBody = Json | FormData;
+export type ModelPropertyNaming = 'camelCase' | 'snake_case' | 'PascalCase' | 'original';
+
+export interface RequestArgs {
+    url: string;
+    options: RequestInit;
+}
+
+export interface RequestOpts {
+    path: string;
+    method: HttpMethod;
+    headers: HttpHeaders;
+    query?: HttpQuery;
+    body?: HttpBody;
+}
+
+export function querystring(params: HttpQuery) {
+    return Object.keys(params)
+        .map((key) => {
+            const value = params[key];
+            if (value instanceof Array) {
+                const multiValue = value.join(`&${encodeURIComponent(key)}=`);
```

Comment:
this only works if the server accepts the same key multiple times (not the case for e.g. php, where it would be `&param[]=value1&param[]=value2`. 
moreover, the values need to be passed through `encodeURIComponent()`, too:
```suggestion
                return value.map(val => `${encodeURIComponent(key)}=${encodeURIComponent(String(val))}`)
                .join('&')
```